cypress/e2e/tests/administration/custom-migration-targets/crud.test.ts

```diff
@@ -224,4 +224,52 @@ describe(["@tier1", "@dc", "@interop"], "Custom Migration Targets CRUD operation
             application.delete();
         });
     });
+
+    Object.values(Languages).forEach((language) => {
+        it(`${language} | custom rule with source technology`, function () {
+            const targetData = this.customMigrationTargets["rules_with_source_element"];
+            const target = new CustomMigrationTarget(
+                data.getRandomWord(8),
+                data.getDescription(),
+                targetData.image,
+                getRulesData(targetData),
+                language,
+                targetData.sources
+            );
+            target.create();
+
+            const application = new Analysis(
+                getRandomApplicationData("bookserverApp", {
+                    sourceData: this.appData["bookserver-app"],
+                }),
+                getRandomAnalysisData(this.analysisData["source_analysis_on_bookserverapp"])
+            );
+            application.create();
+
+            // TC MTA-404
+            Analysis.open();
+            selectItemsPerPage(100);
+            application.selectApplication();
+            cy.contains(button, analyzeButton, { timeout: 20 * SEC })
+                .should("be.enabled")
+                .click();
```

Comment:
This can be implemented with usual `click()` or `clickByText()` button, no?

Reply:
Thanks for bringing this up, @ibragins . Yes, I've tried using clickByText(button, analyzeButton) and the same for the next button. While it succeeds at times, occasionally it fails due to timeout issues. That's why I found it more convenient and reliable to use this approach.